packages/camera/camera_android_camerax/lib/src/image_capture.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart' show BinaryMessenger;
+
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+import 'use_case.dart';
+
+/// Use case for picture taking.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/ImageCapture.
+class ImageCapture extends UseCase {
+  /// Creates a [ImageCapture].
+  ImageCapture(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetFlashMode,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageCaptureHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    _api.createFromInstance(this, targetFlashMode, targetResolution);
+  }
+
+  /// Constructs a [ImageCapture] that is not automatically attached to a native object.
+  ImageCapture.detached(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetFlashMode,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageCaptureHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+  }
+
+  late final ImageCaptureHostApiImpl _api;
+
+  /// Flash mode used to take a picture.
+  final int? targetFlashMode;
+
+  /// Target resolution of the image output from taking a picture.
+  final ResolutionInfo? targetResolution;
+
+  /// Constant for automatic flash mode.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_AUTO().
+  static const int flashModeAuto = 0;
+
+  /// Constant for on flash mode.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_ON().
+  static const int flashModeOn = 1;
+
+  /// Constant for no flash mode.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_OFF().
+  static const int flashModeOff = 2;
+
+  /// Sets the flash mode to use for image capture.
+  void setFlashMode(int newFlashMode) {
+    _api.setFlashModeFromInstance(this, newFlashMode);
+  }
+
+  /// Takes a picture and returns the absolute path of where the capture image
+  /// was saved.
+  Future<String> takePicture() async {
```

Comment:
I really like this implementation of this method, but I have a few notes to add.

This is returning a `String` despite the output being a [OutputFileResult](https://developer.android.com/reference/androidx/camera/core/ImageCapture.OutputFileResults). I'm assuming this is done because it only has one method, `getSavedUri`. But when a library returns a class with a single method, there is a pretty good chance they intend to add more at a later date and they don't want to introduce a breaking change.

This is also missing the [OutputFileOptions](https://developer.android.com/reference/androidx/camera/core/ImageCapture.OutputFileOptions).

But I would guess that the tedious process of wrapping native apis contributed to creating a simpler method. And that's still on me to improve it. So, I would suggest including in the documentation that this method doesn't completely wrap the method from `ImageCapture` and serves as a convenience method that was created for Dart.

Reply:
I didn't wrap those since the goal right now is to reach feature parity, and it was simpler to configure taking a picture the same as [the current plugin](https://github.com/flutter/packages/blob/main/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java#L543) on the Java side than wrap something we aren't likely to change soon.

I think you make a good point though. I'll leave a comment, and I'm definitely open to coming back to wrap those classes later if needed. Should be straightforward to change.